### PR TITLE
Texture thumbnail fixes.

### PR DESCRIPTION
**Pull request by pmuetschard**
- Fixes a bug in the resize code for uncompressed formats
- If luminance channel is requested of a format that doesn't have a luminance channel, but has a single (other) color channel, use that channel as luminance.

### CODE REVIEW

File: core/stream/format.go [hunk 1]
```diff
@@ -129,6 +129,21 @@ func (f *Format) HasColorComponent() bool {
 	return false
 }
 
+// HasSingleColorComponent returns a component if the format contains a single
+// color component, otherwise nil. See ColorChannels for channels considered colors.
+func (f *Format) HasSingleColorComponent() *Component {
```
Comment (ben-clayton) [inline core/stream/format.go]: This signature confuses me. I would expect it to return a bool, iff there's a single color component. Instead it returns the first color component it finds.

File: core/stream/convert.go [hunk 2]
```diff
@@ -113,6 +113,11 @@ func resolveImplicitMappings(count int, mappings []mapping, srcFmt *Format, srcD
 			} else if srcFmt.HasColorComponent() {
 				m.src = buf0
 			}
+		case Channel_Luminance:
+			if c := srcFmt.HasSingleColorComponent(); c != nil {
+				m.src = buf{srcData, c, srcFmt.BitOffsets()[c], uint32(srcFmt.Stride()) * 8}
+			}
+			// TODO: RGB->Luminance conversion
```
Comment (ben-clayton) [inline core/stream/convert.go]: Make it clearer that we're just reusing the first color component we find for luminance. A link to a github issue would also be good.

File: core/stream/format.go (hunk 1 above)
Comment (pmuetschard) [inline core/stream/format.go]: Done.

File: core/stream/convert.go (hunk 2 above)
Comment (pmuetschard) [inline core/stream/convert.go]: Done.

File: core/stream/format.go [hunk 3]
```diff
@@ -129,6 +129,21 @@ func (f *Format) HasColorComponent() bool {
 	return false
 }
 
+// GetSingleColorComponent returns a component if the format contains a single
+// color component, otherwise nil. See ColorChannels for channels considered colors.
+func (f *Format) GetSingleColorComponent() *Component {
```
Comment (ben-clayton) [inline core/stream/format.go]: Can we call this `GetFirstColorComponent`?

Comment (ben-clayton) [inline core/stream/format.go]: Then the comment becomes:

```
// GetFirstColorComponent returns the first color component of the format, or nil if the
// format has no colors. See ColorChannels for channels considered colors.
```

File: core/stream/convert.go [hunk 4]
```diff
@@ -113,6 +113,12 @@ func resolveImplicitMappings(count int, mappings []mapping, srcFmt *Format, srcD
 			} else if srcFmt.HasColorComponent() {
 				m.src = buf0
 			}
+		case Channel_Luminance:
+			if c := srcFmt.GetSingleColorComponent(); c != nil {
+				// A format with a signle color channel is equivalent to a luninance format.
```
Comment (ben-clayton) [inline core/stream/convert.go]: s/signle/single

File: core/stream/format.go (hunk 3 above)
Comment (pmuetschard) [inline core/stream/format.go]: That is not what it does, though, nor what I want it to do. I want it to check whether the format represents a single color channel, not a format that has at least one color channel.

Comment (ben-clayton) [inline core/stream/format.go]: Sorry, I obviously didn't actually read the code.

**Pull request merged**